### ISSUE TITLE
Getting `unsupported URL protocol` when using SSH authentication to clone a private repo

### ISSUE THREAD

**Issue by bulam** [ropensci/git2r#415]
I'm trying to clone a private repo with git2r::clone via SSH in R by doing

`git2r::clone("git@git.server.com:team_name/repo_name.git", "~/dev/")`

but, I keep getting

`Error in 'git2r_clone': unsupported URL protocol`

What could be wrong?

I am authenticated to the repo via SSH and can clone using the same URL in the command line with `git clone url`. I would like to avoid using https and entering my credentials into the URL.

I made sure to install `libssh2`.

**Comment by bulam** [comment]
I found my installation did not have SSH enabled.

Confirmed by running
`libgit2_features()` in R.
which said
```
$threads
[1] FALSE

$https
[1] TRUE

$ssh
[1] FALSE
```
To fix this, I reinstalled `libssh2` and `libgit2`. In my case, I had `libssh2`, but I was still missing `libgit2`. On a Mac, I did

`brew install libssh2` and `brew install libgit2`.

After that, I re-installed git2r from source with

`install.packages("git2r", type="source", configure.vars="autobrew=yes")`

Then, after reloading git2r with `library(git2r)` when I `libgit2_features()` I got `ssh TRUE` and could clone the repo
